google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java

```diff
@@ -138,19 +140,16 @@ private Subscriber(Builder builder) {
     executorProvider = builder.executorProvider;
 
     ExecutorProvider systemExecutorProvider = builder.systemExecutorProvider;
-    if (systemExecutorProvider == null) {
-      systemExecutorProvider =
-          FixedExecutorProvider.create(
-              Executors.newScheduledThreadPool(Math.max(6, 2 * numPullers)));
-    }
-
     alarmsExecutor = systemExecutorProvider.getExecutor();
+
     if (systemExecutorProvider.shouldAutoClose()) {
       closeables.add(
           new AutoCloseable() {
             @Override
             public void close() {
-              alarmsExecutor.shutdown();
+              // Force a shut down, since there are some jobs like the Watcher and MessageDispatcher
+              // that don't shut down cleanly, but don't need to stay on.
+              alarmsExecutor.shutdownNow();
```

Comment:
I think this will cause issue for all inflight messages as well as nacking of any pending message since this same system executor is used to all message.

Reply:
ack.  We'll have to manually shut down a couple of scheduled threads, which will cause some difficulty.  I'll revert the `shutdownNow()`, but keep the thread pool setup for this PR.